csharp/src/Apache.Arrow.Adbc/AdbcConnection.cs

```diff
@@ -0,0 +1,199 @@
+﻿/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Apache.Arrow.Ipc;
+
+namespace Apache.Arrow.Adbc
+{
+    /// <summary>
+    /// Provides methods for query execution, managing prepared statements, using transactions, and so on.
+    /// </summary>
+    public abstract class AdbcConnection : IDisposable
+    {
+        private bool _autoCommit = true;
+        private bool _readOnly = false;
+        private IsolationLevel _isolationLevel = IsolationLevel.Default;

```

Comment:
These might have to be `protected` for implementors?

Reply:
I would prefer to leave them as private. That will force implementors to have their own for getting and setting.